packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -421,6 +424,33 @@ function resolve(
 
     if (shouldConstruct(Component)) {
       inst = new Component(element.props, publicContext, updater);
+
+      if (typeof Component.getDerivedStateFromProps === 'function') {
+        partialState = Component.getDerivedStateFromProps(
+          element.props,
+          inst.state,
+        );
+
+        if (__DEV__) {
+          if (partialState === undefined) {
+            const componentName = getComponentName(Component) || 'Unknown';
+
+            if (!didWarnAboutUndefinedDerivedState[componentName]) {
+              warning(
+                false,
+                '%s.getDerivedStateFromProps(): A valid state object (or null) must be returned. ' +
+                  'You may have returned undefined.',
```

Comment:
Nit: we're sure in this case, maybe drop the "may"

Reply:
Yeah, we used this same wording for the warning from `render`. I assumed the reason was b'c the user may have `return undefined` or they may have just forgotten/omitted the return entirely.

Happy to change it though.